src/LazyLoadImage/LazyLoadImage.tsx

```diff
@@ -14,11 +15,18 @@ import type { LazyLoadImageProps } from "./LazyLoadImage.types";
 export default function LazyLoadImage({
   src,
   alt,
-  ImgProps
+  ImgProps = {}
 }: LazyLoadImageProps) {
   // state to track if the image is visible
   const [isVisible, setIsVisible] = useState(false);
 
+  // states to track the status of the image load, these influence whether to display alt, or MUI Skeleton
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [hasError, setHasError] = useState(false);
+
+  // state variable to set the size for the skeleton to be displayed
+  const [dimensions, setDimensions] = useState({ height: 0, width: 0 });
```

Comment:
The logic in this file is the very same as in the above, so check those comments in doubt. This is the main difference, as we are observing here an image, we can get the size of the image from its reference, which I use to size the `Skeleton`. This had to be handled differently in the other file, as there we are observing the bounding box, which is a flex box.